api/controllers/orderController.js

```diff
@@ -1,25 +1,33 @@
-import orderService from '../services/orderService.js'
+import redis from '../../redis/index.js'
+import config from '../../config/index.js'
+import utils from '../../utils/index.js'
+import glovoService from '../../services/glovoService.js'
+import locationService from '../../services/locationLqService.js'
 
 class OrderController {
   constructor (data) {
     this.data = data
   } // not used yet
 
-  async home (req, res) {
-    // res.render("index", { someInfo: "Data for send" });
-    console.log('?Home page')
+  async oneWay (req, res) {
+    res.json({ result: 'not yet' })
   }
 
-  async oneWayOrder (req, res) {
-    const result = await orderService.glovoApiSend(`${process.env.GLOVO_API_DOMAIN}b2b/orders`, req.method, req.body)
-    res.json({ result })
-  }
-
-  async estimateOrder (req, res) {
-    const result = await orderService.glovoApiSend(`${process.env.GLOVO_API_DOMAIN}b2b/orders/estimate`, req.method, req.body)
-    result.total.discount = { discount: '20%', amount: orderService.getDiscont(result.total.amount) }
-    console.log(result)
-    res.json({ result })
+  async estimate (req, res) {
+    const key = utils.getHash(req.body)
+    let data = await redis.get(key)
+    if (!data) {
+      const locationArr = Object.entries(req.body).map(async ([key, el], index) => {
```

Comment:
You need to expect only { from: string, to: string } request body structure.
So don't need to get location for all req.body, there can be some junk data, which will broke your app.
Example: if I send the req with body { from: 'bla-bla', to: 'bla-bla', destination: 'bla-bla', point: 'bla-bla' }
So your code will make an additional requests to location api with 'destination' and 'point' values.